database/migrations/2021_09_26_164200_create_hrsystem_table.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+
+class CreateHrSystemTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('hrSystem', function (Blueprint $table) {
+            $table->increments('hrSystem_id');
+            $table->unsignedInteger('device_id')->index();
+            $table->string('key', 32);
+            $table->string('value', 32)->nullable();
```

Comment:
This is the wrong database layout. hrSystem contains a fixed number of items.
```
hrSystemUptime
hrSystemDate
hrSystemInitialLoadDevice
hrSystemInitialLoadParameters
hrSystemNumUsers *
hrSystemProcesses *
hrSystemMaxProcesses *
```

Of which, only 3 make sense to store.  Just store them as direct fields in the table.
Also, I don't like _prev fields for any of these.  It only make sense to alert against a max or fixed number for these.
Knowing that the value changed from a previous value is not really helpful.

Reply:
Table changed :-)